### PR TITLE
Refactor code for readability.

### PR DESCRIPTION
**Pull request by stefanbirkner**
## Overview

While reading this class I found parts of the code that required some time for understanding because the code was not as meaningful and/or readable as it could be. I rewrote it into code that is at least better readable for me. Hopefully this helps other readers, too.

---

I hereby agree to the terms of the JUnit Contributor License Agreement.

### CODE REVIEW

**Comment by sbrannen** [comment]
Thanks, @stefanbirkner.

File: junit-launcher/src/main/java/org/junit/gen5/launcher/main/Root.java [hunk 1]
```diff
@@ -44,15 +50,13 @@ TestDescriptor getTestDescriptorFor(TestEngine testEngine) {
 	}
 
 	void applyPostDiscoveryFilters(TestDiscoveryRequest discoveryRequest) {
-		TestDescriptor.Visitor filteringVisitor = (descriptor, remove) -> {
-			if (!descriptor.isTest())
-				return;
-
-			if (isExcluded(discoveryRequest, descriptor)) {
+		Filter<TestDescriptor> postDiscoveryFilter = composeFilters(discoveryRequest.getPostDiscoveryFilters());
```
Comment (dinesh707) [inline junit-launcher/src/main/java/org/junit/gen5/launcher/main/Root.java]: tab the 2nd line

File: junit-launcher/src/main/java/org/junit/gen5/launcher/main/Root.java [hunk 2]
```diff
@@ -60,17 +64,12 @@ void applyPostDiscoveryFilters(TestDiscoveryRequest discoveryRequest) {
...
-		};
-		acceptInAllTestEngines(pruningVisitor);
+		acceptInAllTestEngines(REMOVE_DESCRIPTORS_WITHOUT_TESTS);
 		pruneEmptyTestEngines();
 	}
 
-	private boolean isExcluded(TestDiscoveryRequest discoveryRequest, TestDescriptor descriptor) {
-		return Filter.composeFilters(discoveryRequest.getPostDiscoveryFilters()).filter(descriptor).excluded();
+	private boolean isExcludedTest(TestDescriptor descriptor, Filter<TestDescriptor> postDiscoveryFilter) {
+		return descriptor.isTest() && postDiscoveryFilter.filter(descriptor).excluded();
```
Comment (dinesh707) [inline junit-launcher/src/main/java/org/junit/gen5/launcher/main/Root.java]: postDiscoveryFilter.filter(descriptor) is this guaranteed to be not null ?

Comment (sbrannen) [inline junit-launcher/src/main/java/org/junit/gen5/launcher/main/Root.java]: yes

File: junit-launcher/src/main/java/org/junit/gen5/launcher/main/Root.java (hunk 1 above)
Comment (sbrannen) [inline junit-launcher/src/main/java/org/junit/gen5/launcher/main/Root.java]: @dinesh707, what do you mean by "tab"? There is nothing wrong with the formatting here.

Comment (dinesh707) [inline junit-launcher/src/main/java/org/junit/gen5/launcher/main/Root.java]: Sorry @sbrannen , i think it was my monitor issue. The composeFilters has jumped down to next line. Its a browser formatting issue. (whats the max number of columns ? 100 ? 120?)

Comment (stefanbirkner) [inline junit-launcher/src/main/java/org/junit/gen5/launcher/main/Root.java]: AFAIK the correct formatting is checked by the build. No manual work needed to check this. Am I right, @sbrannen?

Comment (sbrannen) [inline junit-launcher/src/main/java/org/junit/gen5/launcher/main/Root.java]: Column width for Java source code is currently 120: https://github.com/junit-team/junit5/blob/master/src/eclipse/junit-eclipse-formatter-settings.xml#L291

Yes, this is enforced by the build via Spotless.

Comment (sbrannen) [inline junit-launcher/src/main/java/org/junit/gen5/launcher/main/Root.java]: FYI: it's not possible to get the formatting wrong in general. Any violations will result in a broken build... which we will not accept.

Comment (dinesh707) [inline junit-launcher/src/main/java/org/junit/gen5/launcher/main/Root.java]: Thank you @sbrannen.

**Pull request merged**